src/main/perl/Reporter.pm

```diff
@@ -45,6 +45,25 @@ sub _rep_setup
     return $_REP_SETUP;
 }
 
+# Join all arguments into a single string
+# Handles undefined arguments and non-printable garbage
+sub _make_message_string
+{
+    # Ensure that there is no undefined arg: replace by <undef> if any, force stringification otherwise.
+    my @args = map {defined($_) ? "$_" : '<undef>'} @_;
+
+    # Make single string
+    my $string = join('', @args);
+
+    # Replace any non-printable character with a ?
+    # negated character classes in this form are a perlism (perldoc perlrecharclass)
+    # TODO: this will remove non-ascii characters, incl. unicode;
+    #       unless 'use utf8' (and similar) is used. :print: means printable in current encoding
```

Comment:
yes, the caller should always take care of proper encoding. 
`use utf8` is a bit a an odd example, as it is only required if the perl code itself is in utf8, like the example in the unittest. so it should not be enabled by default.